extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,317 @@
+<?php
+
+use Wikia\Logger\WikiaLogger;
+
+class ImageListGetter extends ImageReviewHelperBase {
+
+	private $userId = 0;
+	private $timestamp;
+	private $state;
+	private $order;
+
+	function __construct( int $timestamp, int $state, int $order ) {
+		parent::__construct();
+		$this->userId = $this->wg->user->getId();
+		$this->timestamp = wfTimestamp( TS_DB, $timestamp );
+		$this->state = $state;
+		$this->order = $order;
+	}
+
+	/**
+	 * Get batch of images for Image Review Tool. Images are filtered based on
+	 * state (eg, unreviewed, rejected, questionable), order (last edited, priority),
+	 * and timestamp.
+	 *
+	 * When getting this batch of images, we first check if the timestamp matches
+	 * a previously reviewed batch of images. If so, we return those images. If not,
+	 * we then check to see if there are any outstanding reviews (reviews that this
+	 * reviewer started on, but didn't finished by making a decision on). If so, we
+	 * return those images. If not again, we'll finally create a new review and return
+	 * those images to the user.
+	 */
+	public function getImageList() {
+		$imageList = $this->fetchPreviousReviewFromTimestamp();
+		if ( !empty( $imageList ) ) {
+			return $imageList;
+		}
+
+		$imageList = $this->fetchUnfinishedReview();
+		if ( !empty( $imageList ) ) {
+			return $imageList;
+		}
+
+		return $this->fetchNewReview();
+	}
+
+	private function fetchPreviousReviewFromTimestamp() : array {
+		return $this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "image_review" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)